File: core/src/remove/tanks/game/level/engine/entity/EntityPrefab.java
package remove.tanks.game.level.engine.entity;

import com.google.common.base.MoreObjects;
import remove.tanks.game.level.engine.entity.component.ComponentPrefab;
import remove.tanks.game.level.preload.PreloadData;
import remove.tanks.game.utility.Prefab;

import java.util.Map;

/**
 * @author Mateusz Długosz
 */
public final class EntityPrefab extends Prefab {
    private final PreloadData preloadData;
    private final Map<Class<? extends ComponentPrefab>, ComponentPrefab> componentPrefabs;

    public EntityPrefab(
            PreloadData preloadData,
            Map<Class<? extends ComponentPrefab>, ComponentPrefab> componentPrefabs
    ) {
        this.preloadData = preloadData;
        this.componentPrefabs = componentPrefabs;
    }

    public PreloadData getPreloadData() {
        return preloadData;
    }

    public Map<Class<? extends ComponentPrefab>, ComponentPrefab> getComponentPrefabs() {
        return componentPrefabs;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("preloadData", preloadData)
                .add("componentPrefabs", componentPrefabs)
                .toString();
    }
}

File: tools/docs-generator/lib/level/engine/entity/entity_prefab.py
class EntityPrefab(object):
    def __init__(self, preload_data, component_prefabs):
        self.preload_data = preload_data
        self.component_prefabs = {}
        for component_prefab in component_prefabs:
            self.component_prefabs[component_prefab.__class__.__name__] = component_prefab

    def get_preload_data(self):
        return self.preload_data

    def get_component(self, type):
        if str(type) in self.component_prefabs:
            return self.component_prefabs[str(type)]
        return None

    def get_all_components(self):
        return self.component_prefabs

    def __str__(self):
        return "EntityPrefab(preload_data={}, component_prefabs=[{}])" \
            .format(
                str(self.preload_data),
                ", ".join('{}={}'.format(key, val) for key, val in self.component_prefabs.items())
            )
